addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1060,28 +1077,26 @@ options.registry.menu_data = options.Class.extend({
                         },
                     });
                 }},
-                {text: _t("Edit the menu"), classes: 'btn-primary', click: function () {
-                    this.trigger_up('action_demand', {
-                        actionName: 'edit_menu',
-                        params: [
-                            function () {
-                                var prom = new Promise(function (resolve, reject) {
-                                    self.trigger_up('request_save', {
-                                        onSuccess: resolve,
-                                        onFailure: reject,
-                                    });
-                                });
-                                return prom;
-                            },
-                        ],
-                    });
-                }},
+                {text: _t("Edit the menu"), classes: 'btn-primary', click: () => {this.editMenuCallBack(this.$target)}, close: true},
                 {text: _t("Stay on this page"), close: true}
             ]
         })).open();
     },
 });
 
+options.registry.direct_menu_edit = options.registry.menu_data.extend({
+    editMenuCallBack: function ($target) {
+        const menuId = Number($target.closest('[data-content_menu_id]').attr('data-content_menu_id'));
+        const dialog = new EditMenuDialog(this, {}, menuId);
+        dialog.on('save', this, function () {
+            this.trigger_up('request_save', {
+                reloadEditor: true,
+            });
+        });
+        dialog.open();
+    },
```

Comment:
Why would you need to call EditMenuDialog explicitely for this option and not for the other one? I think you don't need two options. Just only one which handles every case through a call to
```javascript
'action_demand', {
            actionName: 'edit_menu',
```
like in the first option.

So for me this task is about improving the menu_data option to automatically allow editing any menu. Not even a need to change website_event code.

Reply:
@qsm-odoo I just pushed, you were right, just using the existing option on the event menu works fine, although I missed a point in one of the screenshots of the specs (https://www.awesomescreenshot.com/image/4878971/592fd569736f18419d6d6dea55bfa5f4) in which the "mega menu" option should be hidden when opening from the event menu, I think it makes sense to have a different option for this case.

Reply:
@sbuhl Why do you want to remove the mega menu option from events? Did you see a good reason to do it @ThanhDodeurOdoo ?

@ThanhDodeurOdoo Even if there is a reason, I still don't think it needs to be event specific (so no code in event): modify the website option to target the website menus and elements which have the data-menu-id stuff and it will be automatically ok... no? (if SBU still wants to remove mega menu from events, the conditon would simply be "if the menu I edit is different of the website one").

Reply:
@ThanhDodeurOdoo @qsm-odoo 
After reconsidering, it was a mistake to remove the mega menu from events.
I remove it from specs.